UnitySDK/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -34,6 +37,30 @@ public abstract class Brain : ScriptableObject
         public void SubscribeAgentForDecision(Agent agent)
         {
             LazyInitialize();
+            if (m_SensorShapes.Count == 0)
+            {
+                // First agent, save the sensor sizes
+                foreach(var sensor in agent.m_Sensors)
+                {
+                    m_SensorShapes.Add(sensor.GetFloatObservationShape());
+                }
+            }
+            else
+            {
+                // Check for compatibility with the other Agents' sensors
+                // TODO make sure this only checks once per agent
```

Comment:
I added this as a sanity check. Not sure if we should disable it, only enable in Development, or always do it. Hopefully will get easier with Vince's changes.

Reply:
wrap in `#if DEBUG`